suite2p/gui/ensemble_pursuit.py

```diff
@@ -0,0 +1,841 @@
+
+    range: scatter plot range widget.
+
+    epOn: button to compute ensemble pursuit.
+
+    nr_ens_text: text demarcating the number of ensembles to fit.
+
+    n_ens: text box for inputting how many ensembles to fit.
+
+    lam_text: lambda parameter descriptive text label.
+
+    lam_input: text box for setting the value of the parameter lambda.
+
+    box_plot_arrows: arrows custom widget for flipping between box plot pages.
+
+    arrows: widget for flipping through which ensembles a neuron belongs to.
+
+    ens_selector: button for highlighting the cells in the selected ensemble in the main window.
+
+    pc: principal components from PCA.
+
+    data_selector: dropdown for selecting which data to plot against ensembles in the scatter plots.
+
+    enter_ind: text box for inputting which data index to plot against the ensembles.
+
+    widget_lst: list, holds the ensemble heatmap boxes.
+
+    color_lst: list, colors for the ensemble heatmap boxes.
+
+    '''
+    def __init__(self, parent=None):
+        super(EPWindow, self).__init__(parent)
+        pg.setConfigOptions(imageAxisOrder='row-major')
+        self.setGeometry(70,70,1100,900)
+        self.setWindowTitle('Visualize data')
+        self.cwidget = QtGui.QWidget(self)
+        self.setCentralWidget(self.cwidget)
+        self.l0 = QtGui.QGridLayout()
+        #layout = QtGui.QFormLayout()
+        self.cwidget.setLayout(self.l0)
+        #self.p0 = pg.ViewBox(lockAspect=False,name='plot1',border=[100,100,100],invertY=True)
+        # --- cells image
+        self.win = pg.GraphicsLayoutWidget()
+        self.win.move(600,0)
+        self.win.resize(1000,500)
+        self.l0.addWidget(self.win,0,0,14,14)
+        layout = self.win.ci.layout
+
+        #Layout widget for plot_scatter
+        self.sc_plots=pg.GraphicsLayoutWidget()
+        self.l0.addWidget(self.sc_plots,6,0,14,14)
+        #self.l0.addItem(self.sc_plots)
+
+        self.make_top(parent)
+
+        self.make_data_selector(parent)
+
+
+        self.sp=parent.Fbin.copy().T
+
+        self.type_of_plot='boxes'
+
+        if self.type_of_plot=='boxes':
+            self.box_layout = QGridLayout()
+            box_widget = QWidget()
+            box_widget.setLayout(self.box_layout)
+            self.l0.addWidget(box_widget,1,1,4,12)
+
+
+        nt,nn=self.sp.shape
+
+        self.compute_PCA()
+
+        self.ptr=0
+
+        self.box_block_ind=0
+
+        self.box_rng=np.arange(self.box_block_ind*70,(self.box_block_ind+1)*70)
+        self.box_rem=0
+        if self.n_components//70<=self.box_block_ind:
+            self.box_rem=self.n_components%70
+            self.box_rng=np.arange((self.box_block_ind)*70,(self.box_block_ind)*70+self.box_rem)
+
+
+
+    def make_top(self,parent):
+        '''
+        Function that collects together making widgets for the very top of the EP window.
+        '''
+
+        self.epOn = QtGui.QPushButton('compute EnsemblePursuit')
```

Comment:
make this button disabled `if not ENSEMBLEPURSUIT` and in that case add a `QtGui.QLabel('pip install ensemblepursuit')`